Add doc into ml_task

QxiaoQ opened the pull request:
Comment python functions in ml_task . Issue https://github.com/kubeflow/fairing/issues/326
<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/kubeflow/fairing/400)
<!-- Reviewable:end -->

QxiaoQ commented:
/cc @jinchihe

QxiaoQ commented:
/retest

jinchihe commented:
@QxiaoQ please see below info to fix the build error.
```
   ???

<frozen importlib._bootstrap>:967: in _find_and_load_unlocked

    ???

<frozen importlib._bootstrap>:677: in _load_unlocked

    ???

/home/travis/virtualenv/python3.7.1/lib/python3.7/site-packages/_pytest/assertion/rewrite.py:142: in exec_module

    exec(co, module.__dict__)

config_test.py:3: in <module>

    from kubeflow.fairing import config as config

../../kubeflow/fairing/__init__.py:2: in <module>

    from kubeflow.fairing.ml_tasks.tasks import TrainJob, PredictionEndpoint

E     File "/home/travis/build/kubeflow/fairing/kubeflow/fairing/ml_tasks/tasks.py", line 77

E       """Create a train job."""

E                               ^

E   IndentationError: expected an indented block
```

QxiaoQ commented:
/cc @jinchihe ml_task  Done.

QxiaoQ commented:
/retest

jinchihe commented on kubeflow/fairing/ml_tasks/tasks.py:
```diff
@@ -67,29 +67,56 @@ def _build(self):
                  input_files=None, backend=None, pod_spec_mutators=None):
+        """
+        Init the Train Job class.
+        args:
```
@QxiaoQ Could you please update here to use uniform format? Thanks a lot!

QxiaoQ replied:
Format use pyment. Thanks.

QxiaoQ commented:
Redo doc for ml_tasks & notebook use pyment. /cc @jinchihe

jinchihe commented:
/retest

abhi-g commented:
You might want to merge upstream/master in this branch to pull in test fixes

QxiaoQ commented:
> You might want to merge upstream/master in this branch to pull in test fixes

After rebase ,there are still  some task failed

ep #2: ____________________________ test_full_notebook_job ____________________________
Step #2: [gw0] linux -- Python 3.7.5 /usr/local/bin/python
Step #2:
Step #2: capsys = <_pytest.capture.CaptureFixture object at 0x7fee229ca110>
Step #2:
Step #2:     def test_full_notebook_job(capsys):
Step #2:         run_full_notebook_submission(capsys, NOTEBOOK_PATH, 'Hello World',
Step #2: >                                    deployer='job')
Step #2:
Step #2: tests/integration/common/test_full_notebook.py:31:
Step #2: _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
Step #2: tests/integration/common/test_full_notebook.py:24: in run_full_notebook_submission
Step #2:     fairing.config.run()
Step #2: kubeflow/fairing/config.py:129: in run
Step #2:     builder.build()
Step #2: kubeflow/fairing/builders/docker/docker.py:36: in build
Step #2:     self.publish()
Step #2: kubeflow/fairing/builders/docker/docker.py:68: in publish
Step #2:     self._process_stream(line)
Step #2: _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
Step #2:
Step #2: self = <kubeflow.fairing.builders.docker.docker.DockerBuilder object at 0x7fee229ca590>
Step #2: line = b'{"errorDetail":{"message":"unauthorized: Unauthorized access."},"error":"unauthorized: Unauthorized access."}\r\n'
Step #2:
Step #2:     def _process_stream(self, line):
Step #2:         raw = line.decode('utf-8').strip()
Step #2:         lns = raw.split('\n')
Step #2:         for ln in lns:
Step #2:             try:
Step #2:                 ljson = json.loads(ln)
Step #2:                 if ljson.get('error'):
Step #2:                     msg = str(ljson.get('error', ljson))
Step #2:                     logger.error('Build failed: %s', msg)
Step #2: >                   raise Exception('Image build failed: ' + msg)
Step #2: E                   Exception: Image build failed: unauthorized: Unauthorized access.
Step #2:



___________________________ test_full_notebook_tfjob ___________________________
Step #2: [gw1] linux -- Python 3.7.5 /usr/local/bin/python
Step #2:
Step #2: capsys = <_pytest.capture.CaptureFixture object at 0x7fdda7043ed0>
Step #2:
Step #2:     def test_full_notebook_tfjob(capsys):
Step #2:         run_full_notebook_submission(capsys, NOTEBOOK_PATH, 'Hello World',
Step #2: >                                    deployer='tfjob', namespace='kubeflow-fairing')
Step #2:
Step #2: tests/integration/common/test_full_notebook.py:36:
Step #2: _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
Step #2: tests/integration/common/test_full_notebook.py:24: in run_full_notebook_submission
Step #2:     fairing.config.run()
Step #2: kubeflow/fairing/config.py:129: in run
Step #2:     builder.build()
Step #2: kubeflow/fairing/builders/docker/docker.py:36: in build
Step #2:     self.publish()
Step #2: kubeflow/fairing/builders/docker/docker.py:68: in publish
Step #2:     self._process_stream(line)
Step #2: _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
Step #2:
Step #2: self = <kubeflow.fairing.builders.docker.docker.DockerBuilder object at 0x7fdda7043190>
Step #2: line = b'{"errorDetail":{"message":"unauthorized: Unauthorized access."},"error":"unauthorized: Unauthorized access."}\r\n'
Step #2:
Step #2:     def _process_stream(self, line):
Step #2:         raw = line.decode('utf-8').strip()
Step #2:         lns = raw.split('\n')
Step #2:         for ln in lns:
Step #2:             try:
Step #2:                 ljson = json.loads(ln)
Step #2:                 if ljson.get('error'):
Step #2:                     msg = str(ljson.get('error', ljson))
Step #2:                     logger.error('Build failed: %s', msg)
Step #2: >                   raise Exception('Image build failed: ' + msg)
Step #2: E                   Exception: Image build failed: unauthorized: Unauthorized access.
Step #2:
Step #2: kubeflow/fairing/builders/docker/docker.py:79: Exception

abhi-g commented:
Seems one failure is due to 503 from GCS access (possible transient error). The other two are failing on 'Unauthorized access' which shouldn't happen with the workload Identity. Let me check...

abhi-g commented:
/test kubeflow-fairing-presubmit

abhi-g commented:
/test all

abhi-g commented:
/test kubeflow-fairing-presubmit

abhi-g commented:
For some reason the PR is hitting a lot of transient failures. WIll try again in some time...

jinchihe commented:
/retest

abhi-g commented:
/retest

abhi-g commented:
I am a bit bummed by the fact that everytime we run tests for this PR, we have a random failure that is unrelated to the change. This clearly points to the fragility of these tests. I'll do a last try, and then will merge this PR manually, as none of these docstrings should be causing those failures. Thanks for your patience.

abhi-g commented:
/retest

abhi-g commented:
/lgtm
/approve
